### PR TITLE
Add TLS options to support GSuite org data ingestion via LDAP

### PR DESCRIPTION
**Pull request by praphull-purohit**
## Hey, I just made a Pull Request!

This PR adds support for two additional TLS options `certs` and `keys`. This PR is derived from https://github.com/backstage/backstage/pull/13407, but omits the `vendorOverride` part, since that part is already merged via https://github.com/backstage/backstage/pull/13406.

The changes to documentation also included an example for configuring LDAP sync to work with Secure LDAP Service.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-ldap/src/ldap/client.ts [hunk 1]
```diff
@@ -44,9 +46,28 @@ export class LdapClient {
...
+        return {
+          secureContext: tlsLib.createSecureContext({
+            cert: certs,
+            key: keys,
+          }),
+          rejectUnauthorized: tls?.rejectUnauthorized,
+        };
+      }
+      return tls;
+    };
```
Comment (tudi2d) [inline plugins/catalog-backend-module-ldap/src/ldap/client.ts:66]: This can be simplified :)

**Review by tudi2d — COMMENTED** [review]
Thanks for following up with a new PR. Commenting on this for some feedback. Would love to get a opinion from a maintainer if we there are any security concerns with loading the key in this backend module inside the catalog.

File: plugins/catalog-backend-module-ldap/src/ldap/client.ts (hunk 1 above)
Comment (praphull-purohit) [reply plugins/catalog-backend-module-ldap/src/ldap/client.ts]: I've tried to simplify it in next commit. Hopefully that improves it

Comment (tudi2d) [reply plugins/catalog-backend-module-ldap/src/ldap/client.ts]: You could remove some `?` & `undefined` checks by just wrapping the relevant code in a `tls && tls.certs && tls.key` check. If `tls.certs` & `tls.keys` are undefined, `readTLSOptionFile` wouldn't run & therefor wouldn't need to be defined.

Comment (praphull-purohit) [reply plugins/catalog-backend-module-ldap/src/ldap/client.ts]: Thanks. Sorry about the verbosity - I work with JS/TS only for hobby projects so not really adept at writing good quality TS code. But I hope the code is in better shape now

**Review by nuritizra — APPROVED** [review]

File: plugins/catalog-backend-module-ldap/src/ldap/client.ts [hunk 2]
```diff
@@ -44,9 +46,20 @@ export class LdapClient {
     bind?: BindConfig,
     tls?: TLSConfig,
   ): Promise<LdapClient> {
+    const secureContext =
+      tls && tls.certs && tls.keys
+        ? tlsLib.createSecureContext({
+            cert: readFileSync(tls.certs).toString(),
```
Comment (benjdlambert) [inline plugins/catalog-backend-module-ldap/src/ldap/client.ts:52]: Can we use `readFile` from `fs/promises` instead? Little unfortunate to block here, even if it's not a hot path.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks! :pray: just a small nit.

File: plugins/catalog-backend-module-ldap/src/ldap/client.ts (hunk 2 above)
Comment (praphull-purohit) [reply plugins/catalog-backend-module-ldap/src/ldap/client.ts]: Done

**Review by benjdlambert — APPROVED** [review]
nice - thanks for this :tada:

I just simplified it a little bit, but this looks good :tada:

**Pull request merged**